include/ginkgo/core/matrix/dense.hpp

```diff
@@ -35,6 +35,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 #include <initializer_list>
+#include <type_traits>
```

Comment:
nit: Where do you use components from `type_traits`?
You are missing the header `ginkgo/core/base/math.hpp`, which contains the `next_precision` type.
Pretty much all of the matrix formats are missing the inclusion of that header.

Reply:
That's leftover from when I was using `std::conditional` instead of `next_precision`. Good point!